packages/gatsby-link/src/index.js

```diff
@@ -121,8 +134,8 @@ class GatsbyLink extends React.Component {
             e.preventDefault()
             // Is this link pointing to a hash on the same page? If so,
             // just scroll there.
-            const { pathname, hash } = parsePath(this.props.to)
-            if (pathname === window.location.pathname) {
+            const { pathname, hash } = parsePath(to)
+            if (pathname === location.pathname) {
```

Comment:
Used the `location` prop instead of the one on the window object. Probably makes no difference but should be more "correct"